n/node-socks-proxy-agent/node-socks-proxy-agent_ubi_8.5.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash -e
+
+# -----------------------------------------------------------------------------
+#
+# Package          : node-socks-proxy-agent
+# Version          : 6.0.0
+# Source repo      : https://github.com/TooTallNate/node-socks-proxy-agent
+# Tested on        : RHEL 8.5,UBI 8.5
+# Language         : Node
+# Travis-Check     : True
+# Script License   : Apache License, Version 2 or later
+# Maintainer       : Bhagat Singh <Bhagat.singh1@ibm.com>
+#
+# Disclaimer       : This script has been tested in root mode on given
+# ==========         platform using the mentioned version of the package.
+#                    It may not work as expected with newer versions of the
+#                    package and/or distribution. In such case, please
+#                    contact "Maintainer" of this script.
+#   
+# ----------------------------------------------------------------------------
+
+PACKAGE_NAME=node-socks-proxy-agent
+#PACKAGE_VERSION is configurable can be passed as an argument.
+PACKAGE_VERSION=${1:-6.0.0}
+PACKAGE_URL=https://github.com/TooTallNate/node-socks-proxy-agent
+
+yum install -y yum-utils git jq
+
+NODE_VERSION=v12.22.4
+#installing nvm
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash
+source ~/.bashrc
+nvm install $NODE_VERSION
+
+
+OS_NAME=$(cat /etc/os-release | grep ^PRETTY_NAME | cut -d= -f2)
+
+#Check if package exists
+if [ -d "$PACKAGE_NAME" ] ; then
+  rm -rf $PACKAGE_NAME
+  echo "$PACKAGE_NAME  | $PACKAGE_VERSION | $OS_NAME | GitHub | Removed existing package if any"  
+ 
+fi
+ 
+
+if ! git clone $PACKAGE_URL $PACKAGE_NAME; then
+        echo "------------------$PACKAGE_NAME:clone_fails---------------------------------------"
+        echo "$PACKAGE_URL $PACKAGE_NAME"
+        echo "$PACKAGE_NAME  |  $PACKAGE_URL |  $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Clone_Fails"
+        exit 0
+fi
+
+cd  $PACKAGE_NAME
+git checkout $PACKAGE_VERSION
+PACKAGE_VERSION=$(jq -r ".version" package.json)
+# run the test command from test.sh
+
+if ! npm install && npm audit fix && npm audit fix --force; then
+    echo "------------------$PACKAGE_NAME:install_fails-------------------------------------"
+    echo "$PACKAGE_URL $PACKAGE_NAME"
+    echo "$PACKAGE_NAME  |  $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Install_Fails"
+    exit 1
+fi
+
```

Comment:
Adding `npm run build --if-present` here, making tests passing (change taken from [here](https://github.com/TooTallNate/node-socks-proxy-agent/runs/4290509130?check_suite_focus=true).


```
Now using node v12.22.4 (npm v6.14.14)
Creating default alias: default -> v12.22.4
Cloning into 'node-socks-proxy-agent'...
remote: Enumerating objects: 437, done.
remote: Total 437 (delta 0), reused 0 (delta 0), pack-reused 437
Receiving objects: 100% (437/437), 86.03 KiB | 4.53 MiB/s, done.
Resolving deltas: 100% (237/237), done.
Note: switching to '6.0.0'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at 7a628bb 6.0.0

> core-js-pure@3.21.0 postinstall /root/node-socks-proxy-agent/node_modules/core-js-pure
> node -e "try{require('./postinstall')}catch(e){}"

Thank you for using core-js ( https://github.com/zloirock/core-js ) for polyfilling JavaScript standard library!

The project needs your help! Please consider supporting of core-js:
> https://opencollective.com/core-js
> https://patreon.com/zloirock
> https://paypal.me/zloirock
> bitcoin: bc1qlea7544qtsmj2rayg0lthvza9fau63ux0fstcz

Also, the author of core-js ( https://github.com/zloirock ) is looking for a good job -)

npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.3.2 (node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"ppc64"})
npm WARN eslint-config-airbnb@19.0.4 requires a peer of eslint-plugin-react-hooks@^4.3.0 but none is installed. You must install peer dependencies yourself.

added 305 packages from 213 contributors and audited 308 packages in 14.215s

78 packages are looking for funding
  run `npm fund` for details

found 1 low severity vulnerability
  run `npm audit fix` to fix them, or `npm audit` for details

> socks-proxy-agent@6.0.0 prebuild /root/node-socks-proxy-agent
> rimraf dist


> socks-proxy-agent@6.0.0 build /root/node-socks-proxy-agent
> tsc


> socks-proxy-agent@6.0.0 test /root/node-socks-proxy-agent
> mocha --reporter spec



  SocksProxyAgent
    constructor
      ✔ should throw an Error if no "proxy" argument is given
      ✔ should accept a "string" proxy argument
      ✔ should accept a `url.parse()` result object argument
    "http" module
      ✔ should work against an HTTP endpoint
    "https" module
(node:772) [DEP0123] DeprecationWarning: Setting the TLS ServerName to an IP address is not permitted by RFC 6066. This will be ignored in a future version.
      ✔ should work against an HTTPS endpoint


  5 passing (65ms)

------------------node-socks-proxy-agent:install_&_test_both_success-------------------------
https://github.com/TooTallNate/node-socks-proxy-agent node-socks-proxy-agent
node-socks-proxy-agent  |  https://github.com/TooTallNate/node-socks-proxy-agent | 6.0.0 | "Red Hat Enterprise Linux 8.5 (Ootpa)" | GitHub  | Pass |  Both_Install_and_Test_Success
```

Reply:
Added npm run build --if-present before test. All tests passed.